plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/BaseProjectImpl.java

```diff
@@ -588,41 +588,36 @@ protected IStatus run(DBRProgressMonitor monitor) {
-                                    jsonWriter.name(resEntry.getKey());
-                                    jsonWriter.beginObject();
-                                    Map<String, Object> resProps = resEntry.getValue();
-                                    for (Map.Entry<String, Object> propEntry : resProps.entrySet()) {
-                                        jsonWriter.name(propEntry.getKey());
-                                        Object value = propEntry.getValue();
-                                        if (value == null) {
-                                            jsonWriter.nullValue();
-                                        } else if (value instanceof Number) {
-                                            jsonWriter.value((Number) value);
-                                        } else if (value instanceof Boolean) {
-                                            jsonWriter.value((Boolean) value);
-                                        } else {
-                                            jsonWriter.value(CommonUtils.toString(value));
-                                        }
-                                    }
-                                    jsonWriter.endObject();
-                                }
-                                jsonWriter.endObject();
-
-                                jsonWriter.endObject();
-                                jsonWriter.flush();
+                try (Writer mdWriter = Files.newBufferedWriter(mdFile, StandardCharsets.UTF_8);
+                     JsonWriter jsonWriter = METADATA_GSON.newJsonWriter(mdWriter)
+                ) {
+                    jsonWriter.beginObject();
+
+                    jsonWriter.name("resources");
+                    jsonWriter.beginObject();
+                    for (var resEntry : resourceProperties.entrySet()) {
+                        jsonWriter.name(resEntry.getKey());
+                        jsonWriter.beginObject();
+                        Map<String, Object> resProps = resEntry.getValue();
+                        for (var propEntry : resProps.entrySet()) {
+                            jsonWriter.name(propEntry.getKey());
+                            Object value = propEntry.getValue();
+                            if (value == null) {
+                                jsonWriter.nullValue();
+                            } else if (value instanceof Number) {
+                                jsonWriter.value((Number) value);
+                            } else if (value instanceof Boolean) {
+                                jsonWriter.value((Boolean) value);
+                            } else {
+                                jsonWriter.value(CommonUtils.toString(value));
                             }
                         }
+                        jsonWriter.endObject();
                     }
+                    jsonWriter.endObject();
 
+                    jsonWriter.endObject();
+                    jsonWriter.flush();
```

Comment:
This `flush` call is redundant. The `close` call guarantees to flush all pending data before closing the writer.